gen-ai/inference/vllm-rayserve-inf2/vllm-rayserve-deployment.yaml

```diff
@@ -0,0 +1,309 @@
+            MAX_NUM_SEQS: "10"
+            PORT: "8000"
+            NEURON_CC_FLAGS: "-O1"
+            LD_LIBRARY_PATH: "/home/ray/anaconda3/lib:$LD_LIBRARY_PATH"
+            NEURON_CORES: "2"
+        deployments:
+          - name: VLLMDeployment
+            autoscaling_config:
+              min_replicas: 1
+              max_replicas: 3
+              target_num_ongoing_requests_per_replica: 1
+            ray_actor_options:
+              num_cpus: 2
+              resources: {"neuron_cores": 2}
+              runtime_env:
+                env_vars:
+                  LD_LIBRARY_PATH: "/home/ray/anaconda3/lib:$LD_LIBRARY_PATH"
+  rayClusterConfig:
+    rayVersion: '2.32.0'
+    enableInTreeAutoscaling: true
+    headGroupSpec:
+      headService:
+        metadata:
+          name: vllm
+          namespace: vllm
+      rayStartParams:
+        dashboard-host: '0.0.0.0'
+        num-cpus: "0"
+      template:
+        spec:
+          schedulerName: my-scheduler # Correct placement
+          containers:
+          - name: head
+            image: public.ecr.aws/data-on-eks/vllm-ray-neuron-mistral7b:latest
+            imagePullPolicy: Always
+            lifecycle:
+              preStop:
+                exec:
+                  command: ["/bin/sh", "-c", "ray stop"]
+            ports:
+            - containerPort: 6379
+              name: gcs
+            - containerPort: 8265
+              name: dashboard
+            - containerPort: 10001
+              name: client
+            - containerPort: 8000
+              name: serve
+            volumeMounts:
+            - mountPath: /tmp/ray
+              name: ray-logs
+            - name: vllm-script
+              mountPath: /home/ray/vllm_serve.py
+              subPath: vllm_serve.py
+            resources:
+              limits:
+                cpu: "2"
+                memory: "20G"
+              requests:
+                cpu: "2"
+                memory: "20G"
+            env:
+            - name: MODEL_ID
+              value: "NousResearch/Meta-Llama-3-8B-Instruct"
+            - name: PORT
+              value: "8000"
+            - name: LD_LIBRARY_PATH
+              value: "/home/ray/anaconda3/lib:$LD_LIBRARY_PATH"
+          volumes:
+          - name: ray-logs
+            emptyDir: {}
+          - name: vllm-script
+            configMap:
+              name: vllm-serve-script
+          nodeSelector:
+            instanceType: mixed-x86
+            workload: rayhead
+    workerGroupSpecs:
+    - groupName: inf2-group
+      replicas: 1
+      minReplicas: 1
+      maxReplicas: 3
+      rayStartParams:
+        # This setting is critical for inf2/Trn1 node autoscaling with RayServe
+        resources: '"{\"neuron_cores\": 2}"'
+      template:
+        spec:
+          schedulerName: my-scheduler # Correct placement
+          containers:
+          - name: worker
+            image: public.ecr.aws/data-on-eks/vllm-ray-neuron-mistral7b:latest
```

Comment:
@ratnopamc Update this image with new one

Reply:
Sure, will create a new ecr repo and image.

Reply:
@vara-bonthu , Pushed the below changes - 

- Fixed Indentation, whitespacing and formatting of the neuron patch file.
- Created a public repository `vllm-ray2.32.0-inf2-llama3` under data-on-eks and pushed image `vllm-ray2.32.0-inf2-llama3` built with the above patch.
- Updated deployment.yaml to include openai install command.
- Tested autoscaling across nodes, works fine!

Please review.